bukkit/rpk-unconsciousness-bukkit/src/main/kotlin/com/rpkit/unconsciousness/bukkit/listener/PlayerRespawnListener.kt

```diff
@@ -38,9 +38,9 @@ class PlayerRespawnListener(private val plugin: RPKUnconsciousnessBukkit) : List
         val minecraftProfileService = Services[RPKMinecraftProfileService::class.java] ?: return
         val characterService = Services[RPKCharacterService::class.java] ?: return
         val unconsciousnessService = Services[RPKUnconsciousnessService::class.java] ?: return
-        val minecraftProfile = minecraftProfileService.getPreloadedMinecraftProfile(bukkitPlayer) ?: return
-        val character = characterService.getPreloadedActiveCharacter(minecraftProfile) ?: return
-        if (!unconsciousnessService.isUnconscious(character)) return
+        val minecraftProfile = minecraftProfileService.getMinecraftProfile(bukkitPlayer) ?: return
+        val character = characterService.getActiveCharacter(minecraftProfile) ?: return
+        if (!unconsciousnessService.isUnconscious(character).join()) return
```

Comment:
PlayerRespawnEvent is fired synchronously, don't use join.
`character` is a CompletableFuture, maybe use preloaded.